e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+
+    def value_width(self, *, timeout: Timeout = None) -> OptionalStr:
+        return get_el_style(self.loc_container, "width", timeout=timeout)
+
+    def value_height(self, *, timeout: Timeout = None) -> OptionalStr:
+        return get_el_style(self.loc_container, "height", timeout=timeout)
+
+    def value_cols(self, *, timeout: Timeout = None) -> OptionalInt:
+        return int_attr(self.loc, "cols", timeout=timeout)
+
+    def value_rows(self, *, timeout: Timeout = None) -> OptionalInt:
+        return int_attr(self.loc, "rows", timeout=timeout)
+
+    def value_resize(self, *, timeout: Timeout = None) -> typing.Optional[Resize]:
+        ret = str_attr(self.loc, "resize", timeout=timeout)
+        return typing.cast(Resize, ret)
+
+    def expect_value(
+        self,
+        value: TextValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if value is None:
+            value = ""
+        self.expect.to_have_value(value, timeout=timeout)
+
+    def expect_width_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        if value is None:
+            expect_el_style(self.loc_container, "width", None, timeout=timeout)
+            expect_el_style(self.loc, "width", "100%", timeout=timeout)
+        else:
+            expect_el_style(self.loc_container, "width", value, timeout=timeout)
+            expect_el_style(self.loc, "width", None, timeout=timeout)
+
+    def expect_height_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_el_style(self.loc, "height", value, timeout=timeout)
+
+    def expect_cols_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "cols", value=value, timeout=timeout)
+
+    def expect_rows_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "rows", value=value, timeout=timeout)
+
+    def expect_resize_to_have_value(
+        self,
+        value: typing.Union[Resize, None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc, "resize", value=value, timeout=timeout)
+
+
+class _InputSelectBase(
+    _WidthLoc,
+    _InputWithLabel,
+):
+    loc_selected: Locator
+    loc_choices: Locator
+    loc_choice_groups: Locator
+    loc_select: Locator
+
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+        *,
+        select_class: str = "",
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"select#{id}.shiny-bound-input{select_class}",
+            loc_label=f"label[for='{id}']",
+        )
+        # `loc_container` does not need to contain selected items
+        # But we should have `self.loc` be `self.loc_selected`
+        self.loc = self.loc_container.locator("option:checked")
+        # Same value
+        self.loc_selected = self.loc
```

Comment:
`self.loc` should be the `<select>` itself, not the selected choices (which will be `self.loc_selected`).

Reply:
a6a800b